### PR TITLE
add input 'data' (dict) to be merged with GitHub context payload

### PR DESCRIPTION
**Pull request by qrkourier**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -42,3 +42,50 @@ This is a random secret string that is used to provide a data integrity hash the
...
+```
+
+This example results in a top-level dict in the webhook payload.
+
+```bash
+# One way to pass a raw field is to use the GitHub CLI which is pre-installed in all hosted runner VMs
+gh workflow --repo myorg/myrepo run --ref $(git rev-parse --abbrev-ref HEAD) --raw-field my_release_version=1.2.3 send-ziti-webhook.yml
+```
+
+```yaml
```
Comment (qrkourier) [inline README.md:68]: The YAML highlighter is prettier

File: dist/index.js [hunk 2]
```diff
@@ -93,13 +93,26 @@ const zitiHttpRequestData = async (req, buf) => {
   });
 };
 
+function keyValuePairLinesToObj (string) {
```
Comment (qrkourier) [inline dist/index.js:96]: add a function to parse the action's input field `data` as a multi-line string with key=value per line

File: dist/index.js [hunk 3]
```diff
@@ -122,8 +135,10 @@ console.log('Going async...');
       process.exit(-1);
     });
 
-    // Get the JSON webhook payload for the event that triggered the workflow
-    const payload = JSON.stringify(github.context.payload, undefined, 2)
+    // Get the JSON webhook payload for the event that triggered the workflow and merge with extra data dict from action input
+    var extraData = {'data': keyValuePairLinesToObj(extraKeyValuePairLines)}
```
Comment (qrkourier) [inline dist/index.js:139]: embed the extra data object in a dict `data`

File: dist/index.js [hunk 4]
```diff
@@ -122,8 +135,10 @@ console.log('Going async...');
...
+    var payloadData =  Object.assign({}, github.context.payload, extraData);
```
Comment (qrkourier) [inline dist/index.js:140]: merge the GitHub payload dict with the `data` dict

File: action.yml [hunk 5]
```diff
@@ -13,6 +13,8 @@ inputs:
   webhook-secret:
     description: 'Secret for signature hash'
     required: true
+  data:
+    description: 'Arbitrary data to include in payload'
```
Comment (qrkourier) [inline action.yml:17]: add an input to accept one or more key=value pairs, one pair per line

**Review by qrkourier — COMMENTED** [review]
ready for review

**Review by smilindave26 — APPROVED** [review]
Looks good. Please use `npm` to update the package.json version before merging.  You might need to locally build/checkin `dist` update as well (I need to refresh my memory here - if you want to merge and then create a follow-up issue for me to solidify this process, that's fine...)

**Comment by qrkourier** [comment]
@smilindave26 Oh, OK. I think you're saying that I need to reflect my changes from `dist/index.js` in top-level `index.js`, and then run one or more `npm` commands to locally build (combine the SDK with index.js?) which checked-in artifact will be the `dist` sub-directory. I believe I'll need to answer this question to proceed:

What is/are the `npm` command(s) that you would typically use to test/build/install the top-level source files e.g. `index.js`? `npm --help` offers me subcommands like `test` and `install`, but I get a lot of errors for `npm install`. Does this step only work on MacOS in light of the fact that only the Darwin build of the SDK is checked in under `dist`?

I'll add a BUILD.md or CONTRIBUTING.md or both based on what I learn here.

**Comment by smilindave26** [comment]
Take a look at the CI.yml workflow, especially `ncc`, for building dist/

**Pull request merged**